test/e2e/annotations/healthcheck.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package annotations
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"k8s.io/ingress-nginx/test/e2e/framework"
+)
+
+var _ = framework.IngressNginxDescribe("Annotations - HealthCheck", func() {
+	f := framework.NewDefaultFramework("healthcheck")
+
+	BeforeEach(func() {
+		err := f.DisableDynamicConfiguration()
+		Expect(err).NotTo(HaveOccurred())
+
+		err = f.NewEchoDeploymentWithReplicas(2)
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	AfterEach(func() {
+	})
+
+	It("should set upstream-max-fails to 11", func() {
+		host := "healthcheck.foo.com"
+		annotations := map[string]string{
+			"nginx.ingress.kubernetes.io/upstream-max-fails": "11",
+		}
+
+		ing := framework.NewSingleIngress(host, "/", host, f.IngressController.Namespace, "http-svc", 80, &annotations)
+		_, err := f.EnsureIngress(ing)
+
+		Expect(err).NotTo(HaveOccurred())
+		Expect(ing).NotTo(BeNil())
+
+		err = f.WaitForNginxConfiguration(
+			func(server string) bool {
+				return Expect(server).Should(ContainSubstring("max_fails=11 fail_timeout=0;"))
```

Comment:
I'm confused why would server has this setting? Is it not about upstream peers?

Reply:
@ElvinEfendi its using the WaitForNginxConfiguration which looks at the whole config rather than just the server. I should have named (server, string) as (conf, string). Either way looks like this is gonna be deprecated :/